src/awkward/contents/regulararray.py

```diff
@@ -471,8 +471,7 @@ def _getitem_next(
             nexthead, nexttail = ak._slicing.head_tail(tail)
             nextcarry = ak.index.Index64.empty(self._length, index_nplike)
             assert nextcarry.nplike is index_nplike
-            if ak.backend(head) == "cuda":
-                head = int(ak.to_backend(head, backend=self._backend)[0])
+            head = ak._slicing.normalize_integer_like(head)
```

Comment:
The special case for cuda is not needed here anymore.

Reply:
In fact, that special case for CUDA forces a GPU → RAM copy.